host-guest/cb7-b2/setup-ops.py

```diff
@@ -0,0 +1,182 @@
+ligand_atoms = np.arange(126, 156)
+
+################################################################################
+# MAIN
+################################################################################
+
+# Create host-guest test system
+print('Creating host-guest system...')
+from openmmtools.testsystems import HostGuestVacuum
+testsystem = HostGuestVacuum()
+
+# Generate an OpenPathSampling template.
+print('Creating template...')
+template = engine.snapshot_from_testsystem(testsystem)
+
+print('Creating an integrator...')
+integrator = openmm.LangevinIntegrator(300*unit.kelvin, 1.0/unit.picoseconds, 2.0*unit.femtoseconds)
+integrator.setConstraintTolerance(1.0e-6)
+
+print("Selecting a platform...")
+platform = openmm.Platform.getPlatformByName('CPU')
+properties = {'OpenCLPrecision': 'mixed'}
+
+# Create an engine
+print('Creating engine...')
+engine_options = {
+    'n_frames_max': 2000,
+    'platform': 'CPU',
+    'n_steps_per_frame': 10
+}
+engine = engine.Engine(
+    template.topology,
+    testsystem.system,
+    integrator,
+    properties=properties,
+    options=engine_options
+)
+engine.name = 'default'
+
+# Create a hot engine for generating an initial unbinding path
+print('Creating a "hot" engine...')
+integrator_hot = openmm.LangevinIntegrator(900*unit.kelvin, 1.0/unit.picoseconds, 2.0*unit.femtoseconds)
+integrator_hot.setConstraintTolerance(1.0e-6)
+engine_hot = engine.from_new_options(integrator=integrator_hot)
+engine_hot.name = 'hot'
+
+# Select engine mover
+paths.EngineMover.engine = engine
+
+# Create storage
+print('Initializing storage...')
+storage = paths.Storage("host-guest.nc", 'w', template=template)
+print storage.save(engine)
+print storage.save(engine_hot)
+
+#
+platform = engine.simulation.context.getPlatform()
+print platform.getName()
+platform = engine_hot.simulation.context.getPlatform()
+print platform.getName()
+
+# Equilibrate
+engine.current_snapshot = template
+engine.minimize()
+initial_snapshot_cool = engine.current_snapshot
+
+engine_hot.current_snapshot = template
+engine_hot.minimize()
+initial_snapshot_hot = engine_hot.current_snapshot
+
+storage.tag['cool_template'] = engine.current_snapshot
+storage.tag['hot_template'] = engine_hot.current_snapshot
+
+# Define order parameters
+# Compute the closest heavy atom distances between receptor and ligand
+# TODO: How do we restrict this to only residues 0 and 1?
+print(template.topology)
+compute_contacts = paths.MDTrajFunctionCV(
+    name="compute_contacts",
+    cv_scalarize_numpy_singletons=False, # needed because compute_contacts() does not return a single numpy array
+    contacts=[[0,1]],
+    f=md.compute_contacts,
+    topology=template.topology,
+    scheme='closest-heavy',
+    ignore_nonprotein=False,
+    periodic=True
+).with_diskcache()
+storage.save([compute_contacts])
+
+# Create CV states for bound and unbound
+def compute_cv(snapshot, center, compute_contacts):
+    from simtk import unit
```

Comment:
Ah, so this means that the intended behavior is that you can't save code from arbitrary packages? But you can override that with the `safe_modules`?

Interesting question then: what happens if you use that override to save, but then reload from a file in an environment that doesn't have that in `safe_modules`? It raises an error then? (I note that the error is in `to_dict`; is there an equivalent check in `from_dict`? That is the real security concern, right?